main.tex

```diff
@@ -7,28 +7,45 @@ \section{Complex Predicates~vs.~Complex Objects}
 
 % https://github.com/DeMaCS-UNICAL/Angry-HEX/blob/4ea72273518dfde6240ece9dadea927fd1b79c3c/dlv/tactic.dlv#L133
 
-$$ canPush(o_a,o_b) :- \&canPush[objects,hills](o_a,o_b). $$
+$$ canPush(o_a,o_b) :- \&canPush[objects,hills](o_a,o_b). \label{main:rule-1} $$
 
-Intuitively, $canPush(o_a, o_b)$ should be true if the object $o_a$ can push the object $o_b$ (which requires some minimum distance and geometric range).
+Intuitively, $canPush(o_a, o_b)$ should be true if the object $o_a$ can push the object $o_b$.
+To establish the truth of $canPush(o_a, o_b)$ spatial information, like minimum distance and geometric range of the objects, is required; this is obtained by outsourcing the computation to an external source, by means of the external predicate $\&canPush$.
+The required knowledge is given as input to $\&canPush$ by using the predicates $objects$ and $hills$: indeed, $\&canPush$ is a \emph{higher-order predicate}.
 
-Note that $objects$ and $hills$ are predicates, and consequently $\&canPush$ is a higher order predicate. We observe that $objects$ and $hills$ characterize the set of objects present in the current level of the game and the structure of the ground in the virtual environment respectively.
+A useful observation is that the predicates $objects$ and $hills$ both represent lists: $objects$ characterizes the set of objects that are present in the current level, while $hills$ encodes the the morphological structure of the ground of the current level.
+These lists represent a form of knowledge that the agent gains before the reasoning phase.
+Indeed, in \ah the objects and the information about the world scene are first acquired by the \textbf{Vision} component, as explained in Section~\ref{sec:agent_base}; only after this phase, the \textbf{Planning} component is invoked and the reasoning process takes place.
+Thus, for the evaluation of the answer sets wrt.~$canPush$, the elements for which $objects$ and $hills$ do not vary: they are encoded as facts, in the set of assertions \(\mathcal{S}\) added to the program \(\mathcal{P}_{AI}\).
 
-In \ah the reasoning process is only invoked after recognizing all objects in the game environment, so for evaluation answer sets wrt.~$canPush$, the objects for which $objects$ and $hills$ hold do not change. They are facts.
+Once understood that $objects$ and $hills$ are representations of sets, we may pose the following question: would it be possible to replace these predicates with sets?
```

Comment:
Would

Reply:
*would* instead of _may_, is that what you mean?

Reply:
No, I meant a capital W after the colon, since you are starting a new sentence (the question)

Reply:
Ok, done. 😄